presto-main/src/main/java/io/prestosql/server/security/UserMapping.java

```diff
@@ -84,33 +86,57 @@ public UserMappingRules(
         }
     }
 
+    enum Case
+    {
+        KEEP(Function.identity()),
+        LOWER(value -> value.toLowerCase(Locale.ENGLISH)),
+        UPPER(value -> value.toUpperCase(Locale.ENGLISH)),
+        /**/;
+
+        private final Function<String, String> transformation;
+
+        Case(Function<String, String> transformation)
+        {
+            this.transformation = requireNonNull(transformation, "transformation is null");
+        }
+
+        public final String transform(String value)
+        {
+            return transformation.apply(value);
+        }
+    }
+
     public static final class Rule
     {
         private final Pattern pattern;
         private final String user;
         private final boolean allow;
+        private final Case kase;
 
         public Rule(String pattern)
         {
-            this(pattern, "$1", true);
+            this(pattern, "$1", true, Case.KEEP);
         }
 
         @JsonCreator
         public Rule(
                 @JsonProperty("pattern") String pattern,
                 @JsonProperty("user") Optional<String> user,
-                @JsonProperty("allow") Optional<Boolean> allow)
+                @JsonProperty("allow") Optional<Boolean> allow,
+                @JsonProperty("case") Optional<Case> kase)
         {
             this(pattern,
                     requireNonNull(user, "user is null").orElse("$1"),
-                    requireNonNull(allow, "allow is null").orElse(TRUE));
+                    requireNonNull(allow, "allow is null").orElse(TRUE),
+                    requireNonNull(kase, "kase is null").orElse(Case.KEEP));
         }
 
-        public Rule(String pattern, String user, boolean allow)
+        public Rule(String pattern, String user, boolean allow, Case kase)
         {
             this.pattern = Pattern.compile(requireNonNull(pattern, "pattern is null"));
             this.user = requireNonNull(user, "user is null");
             this.allow = allow;
+            this.kase = kase;
```

Comment:
Use require non null

Reply:
I forgot

But anyway, isn't this code too defensive? Using `requireNotNull` everywhere adds a lot of noise to code. Only 4 lines (field declarations) in this class are meaningful, the rest is ugly boilerplate code!

It's enough to use `@NotNull` annotation, IntelliJ recognizes it, and can be configured to produce compile error when you try to to pass a nullable value to `@NotNull` parameter.

Another alternative: use immutables.org annotation processor. It can generate all this mess automatically, eg recognizes `@NotNull` annotation and generates appropriate `requireNonNull` from it.

Reply:
> `@NotNul`

This is also a boilreplate. In general most variables are non null. So I would prefer to configure compiler to fail if `@Nullable` is missing. Is it possible.

Also this is a discussion about project convention which way out of the scope of this pull request.